### DISCUSSION TITLE
schema变更后，表单提交后产出的数据结构参照schema已删除的字段还是仍存在

### DISCUSSION THREAD

**Discussion by youuss** [alibaba/formily#2922] [Q&A 2.x] [answered]
例如本有一个数据结构
``` json
{
  "name": "demo",
  "sex": "name"
}
```
对应的schma搭建表单数据回显为
<img width="628" alt="image" src="https://user-images.githubusercontent.com/34266177/157838715-49f6363c-57b0-4b08-a09e-366896ed0421.png">

调整表单删除性别对应的输入框后为
<img width="631" alt="image" src="https://user-images.githubusercontent.com/34266177/157838905-c5a9ab88-8d0f-4631-8591-c66a572affff.png">

重新提交表单，我期望的最终数据结构为
``` json
{
  "name": "demo1"
}
```

但实际仍保留了删除的性别字段内容
``` json
{
  "name": "demo1",
  "sex": "name"
}
```

想请问有什么具体的文档能够进行form的字段过滤，使得已移除的字段不再会到最终数据结构中？

**Comment by janryWang** [comment, marked as answer]
自己手动过滤

**Reply by youuss** [reply]
我看到之前也有类似的描述，是form对于schema的调整是无感知的。那我应该怎样去获取到已被移除的字段 从而去做手动过滤，或者有类似处理的文档嘛？

**Reply by HuColin** [reply]
如果手动过滤的话，从Schema中删除的字段如果有校验的话，使用submit获取数据的时候，这个字段也会被校验到

**Reply by HuColin** [reply]
> 如果手动过滤的话，从Schema中删除的字段如果有校验的话，使用submit获取数据的时候，这个字段也会被校验到

PS：如果自己过滤的话需要自己解析Schema得到一份最终提交的需要的Key，然后从Form的value里面一一获取吗？
